manager_self_clearing_test.go

```diff
@@ -0,0 +1,89 @@
+package jasper
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func (m *selfClearingProcessManager) registerBasedCreate(ctx context.Context, t *testing.T, opts *CreateOptions) (Process, error) {
```

Comment:
it's a little weird to define methods on a type in the test file, at least to my eyes.

Reply:
I agree. I also felt weird about this, but it helped remove duplication in the test code.